mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/chess/notation/FenNotationTest.kt

```diff
@@ -0,0 +1,184 @@
+package ch.epfl.sdp.mobile.test.application.chess.notation
+
+import ch.epfl.sdp.mobile.application.chess.engine.*
+import ch.epfl.sdp.mobile.application.chess.notation.FenNotation
+import ch.epfl.sdp.mobile.application.chess.notation.FenNotation.parseFen
+import ch.epfl.sdp.mobile.test.application.chess.engine.Games.FoolsMate
+import ch.epfl.sdp.mobile.test.application.chess.engine.play
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+
+class FenNotationTest {
+
+  private fun assertEquivalent(actual: Board<Piece<Color>>?, expected: Board<Piece<Color>>) {
+    for (x: Int in 0 until Board.Size) {
+      for (y: Int in 0 until Board.Size) {
+        assertThat(actual?.get(Position(x, y))?.rank).isEqualTo(expected[Position(x, y)]?.rank)
+        assertThat(actual?.get(Position(x, y))?.color).isEqualTo(expected[Position(x, y)]?.color)
+      }
+    }
+  }
```

Comment:
This could actually be solved using custom subjects in Truth (https://truth.dev/extension.html). Probably not worth doing in this PR, but this should be tracked in a dedicated issue.

Reply:
Nice, did not know about that. Will address it and probably use it in the future parsers to come (yes there are more). #285